backend/src/main/kotlin/com/tngtech/apicenter/backend/domain/service/ServicePersistor.kt

```diff
@@ -3,10 +3,11 @@ package com.tngtech.apicenter.backend.domain.service
 import com.tngtech.apicenter.backend.domain.entity.ResultPage
 import com.tngtech.apicenter.backend.domain.entity.ServiceId
 import com.tngtech.apicenter.backend.domain.entity.Service
+import java.util.*
 
 interface ServicePersistor {
     fun save(service: Service)
-    fun findAll(pageNumber: Int, pageSize: Int): ResultPage<Service>
+    fun findAll(pageNumber: Int, pageSize: Int, username: String, anonymousUsername: String): ResultPage<Service>
```

Comment:
So far, I've seen `jwtAuthenticationProvider.getCurrentUserId()` to get the current user id, a `User` object and now just `username` as a parameter. Could we please unify that? Maybe all methods take a `User` parameter, or all get the user from `jwtAuthenticationProvider`.

Reply:
[Firstly, my mistake, anything called userId is now username, as that is now the id of a user. The Id used to be an auto-generated UUID. I've renamed `userId` where it appeared.] Secondly, I see your point, about consistently representing the idea of a User in the code. However there are sortof two representations of a user: who claims to be authenticated (handled by the token verification, a property of the session) and additional info attached to a certain user (handled by the UserPersistor / User object, which for now only includes an email address). I could perform a findById(), which links the two concepts by the username, but for permissions management, we fundamentally do not need any other information than the (verified) username claim. Thoughts?

Reply:
For now, that makes sense. In general, I could see a `User` class that contains the verified `username` and optionally other stuff as well.